spec/models/loan_health_check_spec.rb

```diff
@@ -121,7 +121,8 @@
           end
 
           context 'ending in 90 days' do
-            let(:loan) { create(:loan, :prospective, end_date: 90.days.from_now) }
+            let(:loan) { create(:loan, :prospective, signing_date: signing_date, end_date: 90.days.from_now) }
+            let(:signing_date) { Date.today }
```

Comment:
I'm a little confused about this. Are you using a variable before it's assigned?

Reply:
Lazy execution of the let's allows this